src/popper/runner_kubernetes.py

```diff
@@ -133,9 +133,12 @@ def _copy_ctx(self):
         the PersistentVolume in the Pod.
         """
         files = os.listdir(self._config.workspace_dir)
-        with tarfile.open(
-            os.path.join(self._config.workspace_dir, "ctx.tar.gz"), mode="w:gz"
-        ) as archive:
+
+        target_file = "ctx_" + str(self._config.wid) + ".tar.gz"
+        source_file = os.path.join(self._config.cache_dir, "kubernetes", target_file)
```

Comment:
When we specify the source as `$POPPER_CACHE_DIR/kubernetes/ctx_abc.tar.gz`, the "kubernetes" directory is not present and the workflow errors out. What we can do is add 2 lines of code to check if the
`$POPPER_CACHE_DIR/kubernetes/` is present or not , and if not present create it.  This can be added at the very top of the `_copy_ctx` function.

Reply:
I just committed with those changes. Let me know what you think!